---
title: ä½¿ç”¨Githubçš„Snakemakeæµç¨‹
author: metseq
date: "2024-03-02"
categories: [Bioinformatics]
toc: true
number-sections: false
---

# æ¦‚è¿°
GitHubæœ‰ä¸€ä¸ª[snakemakeæµç¨‹çš„åˆ†ç±»](https://snakemake.github.io/snakemake-workflow-catalog/)ï¼Œé‡Œé¢æœ‰å¾ˆå¤šç”Ÿä¿¡åˆ†ææµç¨‹ï¼Œrna-seq-star-deseq2ï¼Œdna-seq-gatk-variant-callingã€‚

æˆ‘æµ‹è¯•äº†[rna-seq-star-deseq2](https://github.com/snakemake-workflows/rna-seq-star-deseq2)è¿™ä¸ªæµç¨‹ã€‚

æ¯ä¸ªæµç¨‹éƒ½æœ‰ä½¿ç”¨è¯´æ˜ï¼Œå¦‚æœéƒ½çœ‹è¯´æ˜å¯ä»¥è·‘é€šï¼Œé‚£å°±å¥½äº†ğŸ¥¹

ä¸‹é¢å‚è€ƒ[rna-seq-star-deseq2çš„è¯´æ˜](https://snakemake.github.io/snakemake-workflow-catalog/?usage=snakemake-workflows%2Frna-seq-star-deseq2)ï¼Œä¸€æ­¥ä¸€æ­¥åš.


# Step 1 å®‰è£…snamakeå’Œsnakedeploy

ä½¿ç”¨mambaå®‰è£…snakemeå’Œsnakedeployï¼Œå¦‚æœæ²¡æœ‰mambaï¼Œè‡ªè¡Œæœç´¢å®‰è£…å§.

```shell
$ mamba create -c conda-forge -c bioconda --name snakemake snakemake snakedeploy

$ conda activate snakemake
```

# Step 2 æµç¨‹éƒ¨ç½²

```shell
$ mkdir proj_240302_test_snakemake

$ cd proj_240302_test_snakemake

$ snakedeploy deploy-workflow https://github.com/snakemake-workflows/rna-seq-star-deseq2 . --tag v2.1.0
```

è¿è¡Œåçš„æ–‡ä»¶ç»“æ„ï¼š

```shell
$ tree
.
â”œâ”€â”€ config
â”‚Â Â  â”œâ”€â”€ README.md
â”‚Â Â  â”œâ”€â”€ config.yaml
â”‚Â Â  â”œâ”€â”€ samples.tsv
â”‚Â Â  â””â”€â”€ units.tsv
â””â”€â”€ workflow
    â””â”€â”€ Snakefile
```

å…¶å®å°±ä¸‹äº†configå’Œworkflowä¸¤ä¸ªæ–‡ä»¶å¤¹ï¼Œé‡Œé¢åˆ†åˆ«æ˜¯é…ç½®æ–‡ä»¶å’Œæµç¨‹ã€‚workflowé‡Œé¢åªæœ‰ä¸€ä¸ªSnakefileï¼ŒSnakefileé‡ŒæŠŠgithubä¸Š
rna-seq-star-deseq2çš„æµç¨‹ä½œä¸ºæ¨¡å—å¯¼å…¥äº†ã€‚

# Step 3 é…ç½®æµç¨‹

æµç¨‹é…ç½®æ˜¯æœ€éº»çƒ¦çš„ï¼Œæˆ‘ä¸‹è½½äº†[ä»“åº“](https://github.com/snakemake-workflows/rna-seq-star-deseq2.git)è‡ªå¸¦çš„æµ‹è¯•æ•°æ®ã€‚

æµ‹è¯•æ•°æ®åœ¨`.test`æ–‡ä»¶å¤¹ä¸‹é¢ï¼Œç»“æ„å¦‚ä¸‹ï¼Œåº”è¯¥å¾ˆæ¸…æ¥šäº†ï¼Œ`config_basic`å°±æ˜¯ç®€å•ç‰ˆçš„é…ç½®æ–‡ä»¶ï¼Œ`config_complex`æ–‡ä»¶å¤¹æ˜¯å¤æ‚ç‰ˆçš„é…ç½®æ–‡ä»¶ï¼Œ`ngs-test-data`å°±æ˜¯åŸå§‹æ•°æ®äº†ã€‚

```shell
$ tree .test
.
â”œâ”€â”€ config_basic
â”‚Â Â  â”œâ”€â”€ config.yaml
â”‚Â Â  â”œâ”€â”€ samples.tsv
â”‚Â Â  â””â”€â”€ units.tsv
â”œâ”€â”€ config_complex
â”‚Â Â  â”œâ”€â”€ config.yaml
â”‚Â Â  â”œâ”€â”€ samples.tsv
â”‚Â Â  â””â”€â”€ units.tsv
â””â”€â”€ ngs-test-data
    â””â”€â”€ reads
        â”œâ”€â”€ a.chr21.1.fq
        â”œâ”€â”€ a.chr21.2.fq
        â”œâ”€â”€ a.scerevisiae.1.fq
        â”œâ”€â”€ a.scerevisiae.2.fq
        â”œâ”€â”€ b.chr21.1.fq
        â”œâ”€â”€ b.chr21.2.fq
        â”œâ”€â”€ b.scerevisiae.1.fq
        â”œâ”€â”€ b.scerevisiae.2.fq
        â”œâ”€â”€ c.scerevisiae.1.fq
        â””â”€â”€ c.scerevisiae.2.fq

5 directories, 16 files
```

åˆ æ‰`Step 2 æµç¨‹éƒ¨ç½²`ä¸‹è½½çš„configæ–‡ä»¶ï¼ŒæŠŠ`.test/config_basic`ä¸­çš„é…ç½®æ–‡ä»¶éƒ½æ‹·è´`proj_240302_test_snakemake/config`æ–‡ä»¶å¤¹ä¸‹ï¼Œç„¶åæŠŠ`.test/ngs-test-data`æ–‡ä»¶å¤¹æ‹·è´åˆ°`proj_240302_test_snakemake`æ–‡ä»¶å¤¹ã€‚

ä¿®æ”¹`proj_240302_test_snakemake/config/config.yaml`æ–‡ä»¶çš„sampleså’Œunité…ç½®è·¯å¾„ï¼š

```shell
samples: config/samples.tsv
units: config/units.tsv
```

é…ç½®æœ‰å¾ˆå¤šç»†èŠ‚ï¼Œæ¯”å¦‚åŸå§‹æ•°æ®çš„æ¥å¤´åºåˆ—ï¼Œå·®å¼‚è¡¨è¾¾åˆ†ç»„ä¿¡æ¯ï¼Œæ˜¯å¦æ˜¯é“¾ç‰¹å¼‚æ–‡åº“ç­‰ç­‰ã€‚

è¯·å‚è€ƒé…ç½®æ–‡ä»¶çš„è¯´æ˜å’Œæµç¨‹è‡ªå¸¦çš„é…ç½®è¯´æ˜ã€‚å½“ç„¶`.test`æ–‡ä»¶å¤¹ä¸­çš„é…ç½®æ–‡ä»¶æ ¼å¼å€¼å¾—å‚è€ƒã€‚

# Step 4 è¿è¡Œæµç¨‹

ç»ˆäºå¯ä»¥è¿è¡Œæµç¨‹äº†ï¼š

```shell
$ snakemake --cores all --use-conda
```

æœç„¶æŠ¥é”™äº†ğŸ˜­ï¼š

```shell
NotImplementedError: Remote providers have been replaced by Snakemake storage plugins. Please use the corresponding storage plugin instead (snakemake-storage-plugin-*).
```

Googleä¸€ä¸‹ï¼Œrna-seq-star-deseq2é‡Œæœ‰ä¸€ä¸ª[issues](https://github.com/snakemake-workflows/rna-seq-star-deseq2/issues/71)å°±æ˜¯è¿™ä¸ªé—®é¢˜ã€‚

åŸå› æ˜¯snakemakeæµç¨‹æœ€è¿‘ï¼ˆ2024å¹´3æœˆ3æ—¥ï¼‰æ—¥æ›´æ–°åˆ°ç‰ˆæœ¬8äº†ï¼Œç‰ˆæœ¬8å¼•å…¥äº†pluginåŠŸèƒ½ï¼Œrna-seq-star-deseq2æµç¨‹è¿˜ä¸æ”¯æŒã€‚issuesé‡Œä¹Ÿç»™å‡ºäº†è§£å†³åŠæ³•ï¼Œåˆ é™¤å·²ç»è£…å¥½çš„snakemakeçš„condaç¯å¢ƒï¼Œç„¶åå®‰è£…7.32.4ç‰ˆæœ¬çš„snakemakeã€‚

```shell
$ mamba deactivate

$ mamba remove -n snakemake --all

$ mamba create -c conda-forge -c bioconda -n snakemake7 snakemake=7.32.4 

$ mamba activate snakemake7
```

é‡æ–°é…ç½®å¥½ç¯å¢ƒåï¼Œå†æ¬¡è¿è¡Œæµç¨‹ï¼š

```shell
$ snakemake --cores all --use-conda
```

æµç¨‹ä¼šå…ˆä¸‹è½½ä¾èµ–çš„ç¯å¢ƒï¼Œå‚è€ƒåŸºå› ç»„ï¼ˆæˆ‘ç”¨äº†å¤§çº¦20åˆ†é’Ÿï¼‰ï¼Œç„¶åè¿è¡Œé‡Œé¢å®šä¹‰çš„æ­¥éª¤ã€‚

ç»“æœæ–‡ä»¶åœ¨`results`æ–‡ä»¶å¤¹ä¸‹ï¼š

```shell
$ ll -rth results/
total 120
drwxr-xr-x  14 chenmin  staff   448B Mar  3 00:17 trimmed
drwxr-xr-x   6 chenmin  staff   192B Mar  3 00:20 star
drwxr-xr-x   4 chenmin  staff   128B Mar  3 00:21 qc
drwxr-xr-x   4 chenmin  staff   128B Mar  3 00:22 counts
-rw-r--r--@  1 chenmin  staff    59K Mar  3 00:22 pca.condition.svg
drwxr-xr-x   5 chenmin  staff   160B Mar  3 00:22 deseq2
drwxr-xr-x   5 chenmin  staff   160B Mar  3 00:23 diffexp
```

# Step 5 ç”ŸæˆæŠ¥å‘Š

```shell
$ snakemake --report report.zip
```

è§£å‹report.zipï¼Œæ‰“å¼€report.htmlï¼Œç½‘é¡µå·¦ä¾§æœ‰4ä¸ªé€‰é¡¹ï¼ŒWorkflowæ˜¯æµç¨‹è¿è¡Œçš„æ­¥éª¤ï¼Œè¯´æ˜ï¼›Statisticsç»Ÿè®¡äº†æ¯ä¸€æ­¥è¿è¡Œçš„æ—¶é—´ï¼›Aboutæ˜¯æµç¨‹ç”¨åˆ°è½¯ä»¶çš„ä½¿ç”¨æƒé™è¯´æ˜ï¼›Resultsæ˜¯æœ€å…³å¿ƒçš„ç»“æœäº†ã€‚

æµ‹è¯•å®ŒæˆğŸ‰